### PR TITLE
feat: making multiple request safe

### PR DESCRIPTION
**Pull request by Stefuu**
making multiple request not fail when there is at least one successful bucket is returned

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: src/utils/promise.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const isRejected = (input: PromiseSettledResult<unknown>): input is PromiseRejectedResult =>
```
Comment (tgfassina) [inline src/utils/promise.ts:1]: this one is not being used

File: src/redux/slices/objectsSlice.tsx [hunk 2]
```diff
@@ -32,9 +33,11 @@ export const userGetAllObjects = createAsyncThunk(
 	"bucket/all/content",
 	async ({ bucketDids }: { bucketDids: Array<string> }, thunkAPI) => {
 		try {
-			const bucketObjects = await Promise.all(
+			const results = await Promise.allSettled(
 				bucketDids.map((did) => getObjectsFromBucket(did))
 			)
+			const bucketObjects = results.filter(isFulfilled).map((item) => item.value)
```
Comment (tgfassina) [inline src/redux/slices/objectsSlice.tsx:39]: Do you think it is worth it logging something (as warning maybe) when something doesn't get fulfilled?

File: src/utils/promise.ts (hunk 1 above)
Comment (Stefuu) [reply src/utils/promise.ts:1]: removed

File: src/redux/slices/objectsSlice.tsx (hunk 2 above)
Comment (Stefuu) [reply src/redux/slices/objectsSlice.tsx:39]: Sure, that makes sense

**Pull request merged**